msg_helpers.go

```diff
@@ -408,14 +408,13 @@ Option:
 		off += int(optlen)
 	case EDNS0SUBNET, EDNS0SUBNETDRAFT:
 		e := new(EDNS0_SUBNET)
-		if err := e.unpack(msg[off : off+int(optlen)]); err != nil {
-			return nil, len(msg), err
+		if err := e.unpack(msg[off : off+int(optlen)]); err == nil {
```

Comment:
`err == nil` ? Where are w detecting the subnet is wrong?

Reply:
Hi @miekg, sorry for the late response, if the [unpack](https://github.com/miekg/dns/blob/master/edns.go#L266) function throws an error, we just ignore it

Reply:
Hmm, I'm a bit confused now. If you have a edns client subnet section and the length != 0; this should in theory hold something that is sane, no?

Reply:
I think we should solve #581 in the unpack function, instead of ignoring all errors here?

Reply:
I thought about that too, but in that case we need some flag that tell us to append or not the subnet to the `edns` var, or maybe I am missing something?

Reply:
@miekg sorry, pinging you to see if you have any comment/suggestion about my [last comment](https://github.com/miekg/dns/pull/588#discussion_r153811138)